ratatui/src/widgets/widget_ref.rs

```diff
@@ -82,12 +82,23 @@ pub trait WidgetRef {
     /// Draws the current state of the widget in the given buffer. That is the only method required
     /// to implement a custom widget.
     fn render_ref(&self, area: Rect, buf: &mut Buffer);
+
+    /// Converts the widget into a boxed widget.
+    fn into_boxed(self) -> Box<dyn WidgetRef>
+    where
+        Self: Sized + 'static,
+    {
+        Box::new(self)
+    }
 }
 
 /// This allows you to render a widget by reference.
-impl<W: WidgetRef> Widget for &W {
-    fn render(self, area: Rect, buf: &mut Buffer) {
-        self.render_ref(area, buf);
+impl<W> WidgetRef for &W
+where
+    for<'a> &'a W: Widget,
+{
+    fn render_ref(&self, area: Rect, buf: &mut Buffer) {
+        self.render(area, buf);
```

Comment:
This is the breaking change, which likely affects [a bunch of places](https://github.com/search?q=ratatui+%22fn+render_ref%22+AND+%28NOT+org%3Aratatui%29+AND+%28NOT+is%3Afork%29&type=code) (294 hits), so should definitely be called out in the highlights when we do change notes. Likely we should do a bit of a deep explanation on this when releasing.

https://github.com/search?q=ratatui%20render_ref%20AND%20(NOT%20org%3Aratatui)%20AND%20(NOT%20is%3Afork)&type=code